.editorconfig

```diff
@@ -0,0 +1,87 @@
+# ============================================================================ #
+#
+# @file .editorconfig
+#
+# @brief Sets `shfmt' rules.
+#
+# See all properties at
+# [EditorConfig Properties](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties).
+#
+# They only allow a single newline separator.
+# [Fork or PR upstream](https://github.com/patrickvane/shfmt/blob/master/syntax/printer.go#L464)
+#
+#
+# ---------------------------------------------------------------------------- #
+
+# All files
+# ---------
+
+[*]
+# use "line feed" with no "carriage return" as EOL
+end_of_line              = lf
+trim_trailing_whitespace = true;
+
+
+# ---------------------------------------------------------------------------- #
+
+# Shell files
+# -----------
+
+[*.{sh,bash,bats,zsh}]
+indent_style       = space  # Use spaces not tabs
+indent_size        = 2      # 2 space width
+binary_next_line   = true   # Pipes go at beginning of newline, not end.
+switch_case_indent = true   # Indent cases and bodies
+space_redirects    = true   # Spaces between `>& FILE'
+keep_padding       = true   # avoids realigning tokens by column
+function_next_line = false  # `foo() {' brace is on same line
+never_split        = false  # Use `;'
+
+[*.sh]
+shell_variant = posix
+
+[*.bats]
+shell_variant = bats
+
+[*.bash]
+shell_variant = bash
+
+[*.zsh]
+shell_variant = posix
+
+
+# ---------------------------------------------------------------------------- #
+
+# C/C++
+# -----
+
+[*.{c,cc,cxx,cpp,ipp,h,hh,hxx,hpp}]
+indent_style = space
+indent_size  = 2
+
+
+# ---------------------------------------------------------------------------- #
+
+# [Auto]Makefiles
+# ---------------
+
+[{Makefile,*.mk,*.am}{,.in}]
+indent_style = tab
+indent_size  = 2
+
+
+# ---------------------------------------------------------------------------- #
+
+# Python3
+# -------
+
+[*.py]
+indent_style = tab
```

Comment:
Blocker: Since Python has significant whitespace it's always a good idea to use spaces in Python. Could you explain the intent here?

Reply:
If we ever have more than one or two Python files in the codebase we're likely to use a Python-specific formatter like `black`, so I would probably leave out the Python section entirely.

Reply:
I thought Python had to use tabs. I can delete that block.